src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1556,10 +1893,13 @@ private ArrayTypeSymbol VisitArrayInitializer(BoundArrayCreation node)
             {
                 // collect expressions, conversions and result types
                 (BoundExpression expression, Conversion conversion) = RemoveConversion(expressions[i], includeExplicitConversions: false);
-                expressions[i] = expression;
                 conversions.Add(conversion);
                 var resultType = VisitRvalueWithState(expression);
                 resultTypes.Add(resultType);
+
+                TrackInferredTypesThroughConversions(expressions[i], expression, _visitResult);
```

Comment:
>TrackInferredTypesThroughConversions [](start = 16, length = 36)

Could you leave a PROTOTYPE marker? I suspect the recording of nullability info for conversions will have to be done in ApplyConversion, as it will involve layers with potentially different types and nullabilities.
Also, I noticed that you added a call to `TrackInferredTypesThroughConversions` following some places that use `RemoveConversion`, but not all of them. Is that intentional?

Reply:
No, that's not intentional. If I missed a place, it's because we don't have a test in the entire compiler codebase that uses that path, as I'm verifying that we're visiting all nodes.

---
In reply to: [265706467](https://github.com/dotnet/roslyn/pull/34017#discussion_r265706467) [](ancestors = 265706467)

Reply:
Or that it's handled in some indirect manner.

---
In reply to: [265731608](https://github.com/dotnet/roslyn/pull/34017#discussion_r265731608) [](ancestors = 265731608,265706467)

Reply:
`TrackInferredTypesThroughConversions` already has a prototype marker on it.

---
In reply to: [265731688](https://github.com/dotnet/roslyn/pull/34017#discussion_r265731688) [](ancestors = 265731688,265731608,265706467)